modules/repository/variables.tf

```
@@ -0,0 +1,57 @@
+variable "name" {}
+
+variable "supermarket_name_override" {
+  default = ""
+}
+
+variable "repo_type" {
+  type = string
+  validation {
+    condition     = can(regex("^cookbook|terraform|ide|other$", var.repo_type))
+    error_message = "The repo_type must be cookbook, terraform, ide or other. Case Sensitive."
+  }
+}
+variable "description" {
+  type    = string
+  default = ""
+}
+variable "homepage_url" {
+  type    = string
+  default = ""
+}
+variable "topics" {
+  type    = list(string)
+  default = []
+}
+
+locals {
+  // supermarket_name
+  supermarket_name = var.supermarket_name_override == null ? var.name : var.supermarket_name_override
```

```suggestion
  supermarket_name = var.supermarket_name_override == "" ? var.name : var.supermarket_name_override
```
I think this is the correct value here, unless you set the default value to `null`

---

if a var is optional it comes through as null